Cargo.toml

```diff
@@ -49,7 +49,8 @@ wayland-cursor = {version = "0.28", optional = true}
 tempfile = {version = "3.2", optional = true}
 xkb = {version = "0.2.1", optional = true}
 xkbcommon-sys = {version = "0.7.5", optional = true}
-x11-dl = {version = "2.18.3", optional = true}
+x11-dl = {version = "2.19.1", optional = true}
+libc = {version = "0.2.107", optional = true}
```

Comment:
What is `libc` used for? It would be good if no additional dependencies gets added.

Reply:
On line 120 of x11.rs I'm using it to set the current locale. This is needed for compose keys to work. I do not know why. To verify this, you can compile my branch on Linux (I'm testing on Debian 11), enable a compose key (a web search will show how to do this in whatever desktop environment you use), and then try to type a character with the compose key into the char_callback in my branch (I added a printout of what unicode character the code point corresponds to). If your compose key is right alt then right alt then a then ' types á. With the locale-setting line present, this works as intended. When the locale-setting line is commented out, no character is typed.

I don't think that it's ideal to have the locale set at a global basis like this, however I am not aware of any better solution. It's also worth noting that libc is already pulled in as a recursive/transitive dependency with the current dependency list, so adding it as a direc dependency is almost a no-op.

Reply:
Alright. Fair enough, the reason I'm asking is that I want to try to reduce the big dependency list at some point as stated in #237